### ISSUE TITLE
[REVIEW]: SeismicMesh: Triangular meshing for seismology

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2687]
**Submitting author:** @krober10nd (<a href="http://orcid.org/0000-0002-8470-0359">Keith Roberts</a>)
**Repository:** <a href="https://github.com/krober10nd/SeismicMesh" target ="_blank">https://github.com/krober10nd/SeismicMesh</a>
**Version:** v3.3.1
**Editor:** @meg-simula
**Reviewers:** @nschloe, @jorgensd
**Archive:** <a href="https://doi.org/10.5281/zenodo.4447042" target="_blank">10.5281/zenodo.4447042</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/ba94127ebbd0ca13c841f047fb5077bd/status.svg)](https://joss.theoj.org/papers/ba94127ebbd0ca13c841f047fb5077bd)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/ba94127ebbd0ca13c841f047fb5077bd"><img src="https://joss.theoj.org/papers/ba94127ebbd0ca13c841f047fb5077bd/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/ba94127ebbd0ca13c841f047fb5077bd/status.svg)](https://joss.theoj.org/papers/ba94127ebbd0ca13c841f047fb5077bd)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/krober10nd/SeismicMesh" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@nschloe, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @meg-simula know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/krober10nd/SeismicMesh">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@krober10nd) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/krober10nd/SeismicMesh">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@krober10nd) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @nschloe it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1007/s00366-006-0014-1 is OK
- 10.1016/j.cageo.2007.06.014 is OK
- 10.1145/2998441 is OK
- 10.1007/s00158-018-1950-2 is OK
- 10.1007/s12665-015-4537-x is OK
- 10.1073/pnas.93.4.1591 is OK
- 10.5194/gmd-12-1847-2019 is OK
- 10.1016/j.jcp.2014.01.005 is OK
- 10.1109/SC.2014.86 is OK
- 10.1002/nme.2579 is OK
- 10.1145/2629697 is OK
- 10.1093/gji/ggv380 is OK
- 10.1190/1.3238367 is OK
- 10.1137/S0036144503429121 is OK
- 10.1190/1.1441754 is OK
- 10.1190/1.1437283 is OK

MISSING DOIs

- 10.1007/978-3-642-04319-2_10 may be a valid DOI for title: Perturbing slivers in 3D Delaunay meshes

INVALID DOIs

- None
```

**Comment by meg-simula** [comment]
Issue started by whedon before proper assignment of reviewers, closing this and trying again.

**Comment by arfon** [comment]
@whedon add @jorgensd as reviewer

**Comment by whedon** [comment]
OK, @jorgensd is now a reviewer

**Comment by arfon** [comment]
OK, you should be good to go here now @meg-simula!

**Comment by meg-simula** [comment]
Ok, thanks @arfon! 

@nschloe @jorgensd @krober10nd Thanks for your patience - ready for the review now.

**Comment by krober10nd** [comment]
Thanks could you correct the title please?  It says ERR ..

**Comment by jorgensd** [comment]
@arfon I am not able to tick the checklist boxes above, and when trying to accept the invitation I get that it is expired.

**Comment by jorgensd** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: :point_left:

**Comment by jorgensd** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by jorgensd** [comment]
> @arfon I am not able to tick the checklist boxes above, and when trying to accept the invitation I get that it is expired.

@arfon Im still not able to tick the boxes above (even now that Im assigned as reveiwer), is it possible to resend an invitiation to me?

**Comment by arfon** [comment]
@whedon re-invite @jorgensd as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@jorgensd please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by arfon** [comment]
@jorgensd can you try accepting the invite again now? The invites should work for a week.

**Comment by jorgensd** [comment]
@whedon check repository

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.84  T=0.27 s (264.0 files/s, 33165.5 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          43            947            719           3504
C++                              6            264            153           1327
Markdown                         4            201              0            584
reStructuredText                 7            260            102            272
TeX                              1             22              0            230
YAML                             2              1              1             40
make                             2             14              8             37
DOS Batch                        1              8              1             26
CMake                            1             12              6             25
INI                              2              2              0             24
TOML                             1              0              0              3
-------------------------------------------------------------------------------
SUM:                            70           1731            990           6072
-------------------------------------------------------------------------------


Statistical information for the repository '449e4b4c398ffa4988367928' was
gathered on 2020/10/22.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Keith Roberts                  252         40560          33019           89.34
keith roberts                   43          4785           3723           10.33
krober10                         5           156            119            0.33

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Keith Roberts              6847           16.9          2.6                8.69
krober10                     67           42.9          5.8                7.46
```

**Comment by jorgensd** [comment]
@krober10nd Could you elaborate on your co-authors contributions to this paper/software? By looking at the code contributions, it seems that all commits has been made by you.

**Comment by krober10nd** [comment]
@jorgensd Yes, you are correct. I made the technical contributions/implementations to this work and wrote the paper and I also maintain the software.  Both Rafael and William offered their scientific/professional opinions and we had numerous intellectual discussions which led to new ideas and which I later pursued and implemented in the code stack. Both helped review and edit offline earlier versions of the program despite the fact they have not committed to the repository.

**Comment by nschloe** [comment]
Okay, I'm going over the text now. One thing I noticed is the following
- [ ] "The usage of 11 cores reduces the generation time of this example from 20 minutes to approximately 2 wall-clock minutes". According to the graph, it's more "from 6 minutes". I would also like to see a dashed line in the graph for perfect speed up (or equal scaling in the axes.) 
- [ ] Also a question: Why don't we have perfect speed-up?

**Comment by krober10nd** [comment]
Hey @nschloe

 For the future, I think we should discuss this in the repository with an issue but I will answer this here briefly. We can start an issue and discuss in the repo. 

> "The usage of 11 cores reduces the generation time of this example from 20 minutes to approximately 2 wall-clock minutes". According to the graph, it's more "from 6 minutes". I would also like to see a dashed line in the graph for perfect speed up (or equal scaling in the axes.)

Sure. I can fix this.

> Also a question: Why don't we have perfect speed-up?

There are two reasons: 1) the domain decomposition strategy that I use does not consider load balancing (it simply slices one axis and does not consider the spatial distribution of points) 2) the communication cost becomes competitive to the parallelization speed-up of the Delaunay triangulation.

**Comment by krober10nd** [comment]
Just an updated: I remade the parallel figure, added it to the paper, and updated the text. 

Python axpy in parallel indeed gets in the way. Something always to be improved.

**Comment by jorgensd** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: :point_left:

**Comment by jorgensd** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1007/s00366-006-0014-1 is OK
- 10.1016/j.cageo.2007.06.014 is OK
- 10.1145/2998441 is OK
- 10.1007/s00158-018-1950-2 is OK
- 10.1007/s12665-015-4537-x is OK
- 10.1073/pnas.93.4.1591 is OK
- 10.5194/gmd-12-1847-2019 is OK
- 10.1016/j.jcp.2014.01.005 is OK
- 10.1109/SC.2014.86 is OK
- 10.1002/nme.2579 is OK
- 10.1145/2629697 is OK
- 10.1093/gji/ggv380 is OK
- 10.1190/1.3238367 is OK
- 10.1137/S0036144503429121 is OK
- 10.1190/1.1441754 is OK
- 10.1190/1.1437283 is OK

MISSING DOIs

- 10.1007/978-3-642-04319-2_10 may be a valid DOI for title: Perturbing slivers in 3D Delaunay meshes

INVALID DOIs

- None
```

**Comment by jorgensd** [comment]
@krober10nd It seems to me that the cgal reference in the Performance Comparison chapter is broken. In the article proof above it shows up as:
The 2D/3D serial performance against gmsh and cgal (Alliez et al., 2020, pp. @cgal:r–ctm2–
20b)

**Comment by krober10nd** [comment]
Okay I put in a fix for this citation and removed the RTM acronym  @jorgensd .  I'll update the benchmark figure with hopefully bigger labels and the clarification about the 2D performance of SM.

**Comment by jorgensd** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: :point_left:

**Comment by jorgensd** [comment]
@krober10nd  As whedon is suggesting, you are missing the DOI of the Tournois paper in the references. The DOI is:  10.1007/978-3-642-04319-2_10

**Comment by krober10nd** [comment]
@whedon check references

**Comment by whedon** [comment]
Checking the BibTeX entries failed with the following error: 

 Failed to parse BibTeX on value "doi" (NAME) [#<BibTeX::Bibliography data=[20]>, "@", #<BibTeX::Entry >, {:title=>["Perturbing slivers in 3D Delaunay meshes"], :author=>["Tournois, Jane and Srinivasan, Rahul and Alliez, Pierre"], :booktitle=>["Proceedings of the 18th international meshing roundtable"], :pages=>["157--173"], :year=>["2009"], :publisher=>["Springer"]}]

**Comment by krober10nd** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1007/s00366-006-0014-1 is OK
- 10.1016/j.cageo.2007.06.014 is OK
- 10.1145/2998441 is OK
- 10.1007/s00158-018-1950-2 is OK
- 10.1007/s12665-015-4537-x is OK
- 10.1073/pnas.93.4.1591 is OK
- 10.5194/gmd-12-1847-2019 is OK
- 10.1016/j.jcp.2014.01.005 is OK
- 10.1109/SC.2014.86 is OK
- 10.1002/nme.2579 is OK
- 10.1145/2629697 is OK
- 10.1093/gji/ggv380 is OK
- 10.1190/1.3238367 is OK
- 10.1137/S0036144503429121 is OK
- 10.1190/1.1441754 is OK
- 10.1007/978-3-642-04319-2_10 is OK
- 10.1190/1.1437283 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by meg-simula** [comment]
Hi @krober10nd @nschloe @jorgensd Just checking in to see how you are doing - let me know if there is anything I can contribute with in this on-going review. Thanks again for your efforts.

**Comment by krober10nd** [comment]
Thanks @meg-simula I've been busy improving the paper incorporating the reviewers suggestions. The reviews have been very productive in my opinion.

**Comment by nschloe** [comment]
I agree. This is still an ongoing effort.

**Comment by krober10nd** [comment]
@nschloe @jorgensd Can you please elaborate more on what ongoing work needs to be done for to finish this review? 

Just to state some recent activity: I've fixed a broken reference in the paper and edited a statement regarding mesh quality on KSP performance. I've improved both the serial and parallel performance comparisons and at least the serial performance comparisons have been corroborated via Nico's meshgen comparison repository. 

Other than that, I've also improved the README to follow the guidelines of the JOSS review by adding a contributing section and clarified the statement of need. 

Thank you

**Comment by jorgensd** [comment]
> @nschloe @jorgensd Can you please elaborate more on what ongoing work needs to be done for to finish this review?
> 
> Just to state some recent activity: I've fixed a broken reference in the paper and edited a statement regarding mesh quality on KSP performance. I've improved both the serial and parallel performance comparisons and at least the serial performance comparisons have been corroborated via Nico's meshgen comparison repository.
> 
> Other than that, I've also improved the README to follow the guidelines of the JOSS review by adding a contributing section and clarified the statement of need.
> 
> Thank you

I think we are getting quite close to a finalized version. I've added some more notes to the various issues in SeismicMesh (and some of them you have already addressed). When we iron out the last few things, I am happy with the state of the software. The final thing would be to have a last read-through of the paper. Nico has given you some good input, and I believe that I will have very few corrections after the thread https://github.com/krober10nd/SeismicMesh/issues/139 is finalized.

**Comment by krober10nd** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: :point_left:

**Comment by krober10nd** [comment]
I fixed the CGAL reference above. Now there's a typsetting issue remaining.

  I made the benchmark figure quite long to deal with the fact that the JOSS article format has large margins but now the caption of that figure is all jumbled up.

I guess the solution is to make the figure wider now.

**Comment by krober10nd** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: :point_left:

**Comment by jorgensd** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: :point_left:

**Comment by krober10nd** [comment]
I would like to check with @nschloe if you have any more comments and suggestions for revision to move this forward. Thank you.

**Comment by meg-simula** [comment]
@nschloe: How is the final bits of this review going? Any additional comments/issues to be resolved?

@jorgensd Thanks for your efforts on this - I note that all your boxes are checked - is this good to go from your side?

**Comment by nschloe** [comment]
> How is the final bits of this review going

I will check as soon as I have time.

**Comment by jorgensd** [comment]
> @nschloe: How is the final bits of this review going? Any additional comments/issues to be resolved?
> 
> 
> 
> @jorgensd Thanks for your efforts on this - I note that all your boxes are checked - is this good to go from your side? 

This is good to go from my side.

**Comment by krober10nd** [comment]
I've dealt with one more issue pertaining to the paper (https://github.com/krober10nd/SeismicMesh/pull/170) @nschloe brought up and I've independently improved parallel performance in 2d/3d by a significant margin in https://github.com/krober10nd/SeismicMesh/pull/171 and  https://github.com/krober10nd/SeismicMesh/pull/172 and updated the figure and text accordingly regarding 3d parallel performance. 

I'd like to know please where this paper stands.

Thanks

**Comment by nschloe** [comment]
I might have time to look at the revisions later tonight, otherwise tomorrow.

**Comment by nschloe** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: :point_left:

**Comment by nschloe** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: :point_left:

**Comment by nschloe** [comment]
This is good to go from my point of view.

Just for the record, let me add a few remarks.

 * @krober10nd was exceptionally responsive to the remarks, and I think the software and the article have greatly improved in the process.
 * I'm not a big fan of how the software targets seismologists only. SeismicMesh is a parallel implementation of distmesh, and as such is general-purpose mesh generator. There is a function that converts seismology data into a density function, but that part is actually separate and can be used with any other mesh generator. I also wouldn't like to see another mesh generator next year that only adapts that small part for medicine and calls it _CellMesh_.
 * For anyone interested, as part of the review process, I compiled a more extensive mesh generator comparison at https://github.com/nschloe/meshgen-comparison (including SeismicMesh).
 * I love the JOSS review process and I'll hopefully start writing up some articles for it myself next year.

Happy holidays everyone!

**Comment by krober10nd** [comment]
Thank you Nico! It has been a great pleasure to work on this with these reviewers. I agree with your comments as well. 

I've learned a tremendous amount from you guys and I really enjoyed this review process especially the open nature of it! Look forward to seeing your articles appear on here @nschloe!

**Comment by krober10nd** [comment]
@whedon generate pdf  

just running Nico's `betterbib` package to add as much information to the papers bibliography as possible.

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: :point_left:

**Comment by meg-simula** [comment]
Early next week will have to be first thing after Jan 4 2021 - Happy Holidays to all of you in the meanwhile!

**Comment by krober10nd** [comment]
Hope the time off was relaxing! Please let me know if there’s anything else to be done. Thank you. @meg-simula

**Comment by krober10nd** [comment]
Hi @meg-simula any updates on this process?

**Comment by meg-simula** [comment]
@whedon generate pdf

**Comment by meg-simula** [comment]
@whedon check references

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: :point_left:

**Comment by meg-simula** [comment]
@krober10nd Happy New Year! Thanks for your patience and nudging here. 

In the paper, could you please address the following minor comment:

- [ ] Core functionality - item 2. This sentence is unclear (something about the "in either" bit), please improve.

**Comment by krober10nd** [comment]
Hey @meg-simula ...Okay, I edited the text in the paper in the Core functionality- item 2 (it was a typo). 

I then made a tagged release V3.3.1 https://github.com/krober10nd/SeismicMesh/releases/tag/V3.3.1 and archived it in Zenodo and edited the metadata on Zenodo to reflect and match the paper. https://zenodo.org/record/4447042#.YAWQvS1h1pQ

The doi is 10.5281/zenodo.4447042

**Comment by meg-simula** [comment]
@whedon set  10.5281/zenodo.4447042 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4447042" target="_blank">10.5281/zenodo.4447042</a> is the archive.

**Comment by meg-simula** [comment]
@whedon set v3.3.1 as version

**Comment by whedon** [comment]
OK. v3.3.1 is the version.

**Comment by meg-simula** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2030

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2030, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1142/9789811202124_0004 is OK
- 10.1007/s00366-006-0014-1 is OK
- 10.1016/j.cageo.2007.06.014 is OK
- 10.1145/2998441 is OK
- 10.1007/s00158-018-1950-2 is OK
- 10.1007/s12665-015-4537-x is OK
- 10.1073/pnas.93.4.1591 is OK
- 10.5194/gmd-12-1847-2019 is OK
- 10.1016/j.jcp.2014.01.005 is OK
- 10.1142/9789811202124_0004 is OK
- 10.1109/sc.2014.86 is OK
- 10.1002/nme.2579 is OK
- 10.1145/2629697 is OK
- 10.1093/gji/ggv380 is OK
- 10.1190/1.3238367 is OK
- 10.1137/s0036144503429121 is OK
- 10.1190/1.1441754 is OK
- 10.1007/978-3-642-04319-2_10 is OK
- 10.1190/1.1437283 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by krober10nd** [comment]
Thanks! Does the chief editor have to approve this now or is good to go?

**Comment by meg-simula** [comment]
This is now with the Editor(s) in Chief, yes.

--
Marie

**Comment by kthyng** [comment]
@krober10nd Hi! I'll be taking over from here. I see your Zenodo archive is set up, and the version of your software is set. 

Paper comments:
* Your #1 under Core Functionality — the last reference should not have ().
* your #3 — I'm outside this field but in ocean meshes, 2d meshes can have bad quality shape (e.g. really narrow triangles). Could you state briefly why 2d meshes don't have degenerate elements?
* Parallelism first sentence: no () around reference.

**Comment by krober10nd** [comment]
@kthyng Thanks! I fixed up the two citations to be in-text rather than parenthetical. In regard to your comment about degenerate element quality, indeed I meant so-called sliver elements which occur only in 3D meshes. I edited the text minimally to be more specific what I meant by degenerate elements in this case.

**Comment by kthyng** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02687/joss.02687/10.21105.joss.02687.pdf) :page_facing_up: :point_left:

**Comment by kthyng** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1351586210758725633) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2036
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02687](https://doi.org/10.21105/joss.02687)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kthyng** [comment]
Congratulations on your new publication @krober10nd! Thanks to editor @meg-simula and reviewers @nschloe and @jorgensd for your time and expertise!!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02687/status.svg)](https://doi.org/10.21105/joss.02687)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02687">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02687/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02687/status.svg
   :target: https://doi.org/10.21105/joss.02687
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02687/status.svg)](https://doi.org/10.21105/joss.02687)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
